rice/Enum.ipp

```diff
@@ -84,17 +84,24 @@ namespace Rice
     // Add comparable support
     klass.include_module(rb_mComparable);
 
-    // Singleton methods
-    klass.define_singleton_method("each", [](VALUE klass)
+        // Singleton methods
+    klass.define_singleton_function("each", [klass]()
         {
-          for (auto& pair : valuesToNames_)
-          {
-            Enum_T enumValue = pair.first;
-            VALUE value = detail::To_Ruby<Enum_T>().convert(enumValue);
-            detail::protect(rb_yield, value);
-          }
+//        Get rid of LocalJumpErrors
+            if (detail::protect(rb_block_given_p))
```

Comment:
This probably doesn't need to be protected as there's no required object conversion going on, but you're also not calling `rb_block_given_p` here (missing the parens) so I'm not sure this even compiles?.

Reply:
Well, it compiles. Isn't it how `detail::protect` is supposed to work? I mean, it accepts a function pointer and calls it.
But now I see it's not required, because no ruby exception can be raised in `rb_block_given_p`.

Reply:
Sorry, you're right, been a while since I've looked at this code.